vm/src/vm/runners/cairo_runner.rs

```diff
@@ -446,19 +446,18 @@ impl CairoRunner {
                 .map_err(RunnerError::MemoryInitializationError)?;
 
             // Mark all addresses from the program segment as accessed
-            let base = self
-                .program_base
-                .unwrap_or_else(|| Relocatable::from((0, 0)));
             for i in 0..self.program.shared_program_data.data.len() {
-                vm.segments.memory.mark_as_accessed((base + i)?);
+                vm.segments.memory.mark_as_accessed((prog_base + i)?);
             }
+        } else {
+            return Err(RunnerError::NoProgBase);
         }
         if let Some(exec_base) = self.execution_base {
             vm.segments
                 .load_data(exec_base, &stack)
                 .map_err(RunnerError::MemoryInitializationError)?;
         } else {
-            return Err(RunnerError::NoProgBase);
+            return Err(RunnerError::NoExecBase);
```

Comment:
Given both return errors if `None`, another implementation could be:
```rust
let prog_base = self.program_base.ok_or(RunnerError::NoProgBase)?;
let exec_base = self.execution_base.ok_or(RunnerError::NoExecBase)?;
// And then the logic without extra indentation or error handling.
```

Reply:
That said, do those always error out? Maybe it could make sense to make them not optional and fail earlier if they're missing?